org.mwc.debrief.pepys/src/main/java/org/mwc/debrief/pepys/model/db/config/ConfigurationReader.java

```diff
@@ -0,0 +1,120 @@
+package org.mwc.debrief.pepys.model.db.config;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Scanner;
+
+import junit.framework.TestCase;
+
+public class ConfigurationReader {
+
+	public static class ConfigurationReaderTest extends TestCase {
+
+		public void testConfigurationRead() {
+			final String testConfiguration = "[database]\n" + "db_username = abcdef\n" + "db_password = _abcdef_\n"
+					+ "db_host = localhost\n" + "db_port = 5432\n" + "db_name = pepys2\n" + "[archive]\n"
+					+ "user = abcdef\n" + "password = abcdef";
+			final InputStream stream = new ByteArrayInputStream(testConfiguration.getBytes());
+			final DatabaseConfiguration databaseConfigurationTest = new DatabaseConfiguration();
+			parseConfigurationFile(databaseConfigurationTest, stream);
+
+			// Check Database Categories
+			final HashMap<String, String> databaseCategory = databaseConfigurationTest.getCategory("database");
+			final HashMap<String, String> archiveCategory = databaseConfigurationTest.getCategory("archive");
+
+			assertTrue("databaseCategory - size", databaseCategory.size() == 5);
+			assertTrue("archiveCategory - size", archiveCategory.size() == 2);
+
+			assertTrue("DBUsername", databaseCategory.get("db_username").equals("abcdef"));
+			assertTrue("DBPassword", databaseCategory.get("db_password").equals("abcdef"));
+			assertTrue("DBHost", databaseCategory.get("db_host").equals("localhost"));
+			assertTrue("DBPort", databaseCategory.get("db_port").equals("5432"));
+			assertTrue("archiveCategory - User", archiveCategory.get("user").equals("abcdef"));
+
+		}
+
+		public void testProcess() {
+			assertTrue("Process Test", process("ABZabz").equals("NOMnom"));
+		}
+	}
+
+	static final String DEMILITER = "_";
+
+	static final String CATEGORY_START = "[";
+
+	static final String CATEGORY_END = "]";
+
+	private static final String CONFIG_SEPARATOR = "=";
+
+	private static final String CONFIG_COMMENT_CHAR = "#";
+
```

Comment:
We have quite a lot of code to handle `ini` files.  

Would this method be an alternate solution? https://stackoverflow.com/a/41084504/92441